includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -815,6 +819,7 @@ public function process_order_for_confirmed_intent( $order, $intent_id, $save_pa
 		}
 		$this->save_intent_to_order( $order, $intent );
 		$this->set_payment_method_title_for_order( $order, $payment_method_type, $payment_method_details );
+		$this->save_order_redirect_processed( $order );
```

Comment:
Each of these methods calls $order->save() which would do a db update statement. Ideally we would only call save() once. Refactoring that is probably too much effort right now but maybe it would be better to not use a separate method and just do 
```suggestion
		$this->save_intent_to_order( $order, $intent );
		$this->set_payment_method_title_for_order( $order, $payment_method_type, $payment_method_details );
		$order->update_meta_data( '_stripe_upe_redirect_processed', true );
		$order->save();
```